components/camera/platforms/jetson/camera.go

```diff
@@ -0,0 +1,139 @@
+package jetsoncamera
+
+// #include <sys/utsname.h>
+// #include <string.h>
+import "C"
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"syscall"
+	"unsafe"
+)
+
+// DetectOSInformation pulls relevant OS attributes as an OSInformation struct
+// Kernel and Device will be "unknown" if unable to retrieve info from the filesystem
+// returns an error if kernel version or device name is unavailable
+func DetectOSInformation() (OSInformation, error) {
+	kernelVersion, err := getKernelVersion()
+	if err != nil {
+		return OSInformation{}, fmt.Errorf("failed to get kernel version: %w", err)
+	}
+	deviceName, err := getDeviceName()
+	if err != nil {
+		return OSInformation{}, fmt.Errorf("failed to get device name: %w", err)
+	}
+	osInfo := OSInformation{
+		Name:   runtime.GOOS,
+		Arch:   runtime.GOARCH,
+		Kernel: kernelVersion,
+		Device: deviceName,
+	}
+	return osInfo, nil
+}
+
+// getKernelVersion returns the Linux kernel version ($ uname -r)
+func getKernelVersion() (string, error) {
+	var utsName C.struct_utsname
+	if C.uname(&utsName) == -1 {
+		return Unknown, fmt.Errorf("uname information unavailable (%v)", utsName)
+	}
+	release := C.GoString((*C.char)(unsafe.Pointer(&utsName.release[0])))
+	return release, nil
+}
+
+// getDeviceName returns the model name of the device
+// ($ cat /sys/firmware/devicetree/base/model)
+func getDeviceName() (string, error) {
+	devicePath := "/sys/firmware/devicetree/base/model"
+	device, err := os.ReadFile(devicePath)
+	if err != nil {
+		return Unknown, err
+	}
+	return string(bytes.TrimRight(device, "\x00")), nil
+}
+
+// DetectError checks if the daughterboard and driver are supported and installed on the device
```

Comment:
I'd rephrase this: if everything is supported and installed, we don't merely check that it looks alright. It's closer to "DetectError returns our best guess of what is wrong on an unsuccessful setup" or something like that.